pkg/autoscaler/config/config.go

```diff
@@ -14,15 +14,14 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package autoscaler
+package config
```

Comment:
I'm guessing this move was prompted by a dependency cycle introduced by referencing the config from the types?

I just want to confirm, since this is one of my concerns about introducing config complexity to the types.

Reply:
Yes, exactly.

Reply:
Generally, I want to provide configuration to the webhook.  I couldn't see another way with the webhook being so abstracted.  Any other ideas?

Reply:
@mattmoor I think this is the only outstanding conversation.  Is this okay?  Need your approval.

Reply:
I posted a suggestion that avoids the global [here](https://github.com/knative/pkg/issues/306#issuecomment-470969536) by threading through `context.Context` and using @dprotaso's approach of attaching a config store to `ctx`.

Since I feel like this introduces that debt, I'd like to understand who owns paying it :)

Reply:
Yeah, it's not a pretty hack.  I wouldn't mind implementing the ctx threading to get rid of it.  Can you give me some more pointers to @dprotaso 's approach?  I couldn't seem to find a PR or issue related to that.

There are now three follow-ups (promises) for this pull request:
1. consolidate / simplify target calculations, addressed in https://github.com/knative/serving/pull/3373
2. implement podautoscaler class defaulting: https://github.com/knative/serving/issues/3415
3. threading through podautoscaler defaults in context to remove the global hack

But I would like to do them in separate PRs because this one is 'outta control.  ;)

Reply:
The configstore stuff snapshots the current state of the config onto the context, so what we access during Reconcile doesn't change mid-reconcile.  Here's [an example](https://github.com/knative/serving/blob/9d024f83a5540fdee3ed75fcaa98cf51f5cd15ce/pkg/reconciler/v1alpha1/revision/revision.go#L251) of that snapshot for the Revision controller.

Reply:
That's cool.  If we want to default in the webhook, this would be the way to do it.  But I'm punting on that now and just defaulting in the controller.